src/extension.ts

```diff
@@ -1137,81 +1137,86 @@ export async function activate(context: vscode.ExtensionContext) {
 						);
 					}
 
-					const rawPath =
-						(uri || vscode.window.activeTextEditor?.document.uri)
-							?.path ?? null;
+					const uri =
+						(uriArg ||
+							vscode.window.activeTextEditor?.document.uri) ??
+						null;
 
-					if (rawPath === null) {
+					if (uri === null || !uri.path) {
 						return;
 					}
 
-					const mostRecentCodemodHash =
-						codemodListWebviewProvider.getMostRecentCodemodHash();
+					const codemodList = await engineService.getCodemodList();
 
-					if (mostRecentCodemodHash === null) {
-						return;
-					}
+					// order: least recent to most recent
+					const top3RecentCodemodHashes = codemodListWebviewProvider
+						.getRecentCodemodHashes()
```

Comment:
I would limit the number of the most recent codemod hashes to 3 already when saving the codemod hashes, because that structure can become infinite (or just quite long) over time.